security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -492,14 +492,31 @@ func sdsDiscoveryResponse(s *model.SecretItem, conID, resourceName string) (*xds
 		Name: s.ResourceName,
 	}
 	if s.RootCert != nil {
-		secret.Type = &authapi.Secret_ValidationContext{
-			ValidationContext: &authapi.CertificateValidationContext{
-				TrustedCa: &core.DataSource{
-					Specifier: &core.DataSource_InlineBytes{
-						InlineBytes: s.RootCert,
+		if s.CRL != nil {
```

Comment:
you can handle secret.Type = {} // without CRL first.

if crl != nil {
 // add crl
}
reduce the nesting level

Reply:
Changed to handle without CRL first, then added crl if non nil.  Did it at the `validationContext` level and then set the `secret.Type` at the end.  Hope that's ok, if not let me know and I can change.